### PR TITLE
Fix: missing webp images

### PR DESCRIPTION
**Pull request by samijaber**
Depends on #90

The diff here will be a lot more legible once #90 is merged into `main`.

Main feature changes:
- supports webp images by re-encoding them as PNGs
- improves handling of `picture` elements by using `img.currentSrc` instead of the first `source` element's `src`

### CODE REVIEW

File: plugin/functions/encode-images.ts [hunk 1]
```diff
@@ -0,0 +1,54 @@
+/**
+ * Figma's `figma.createImage()` only accepts PNG, JPEG and GIF. We therefore need to transform webp images.
+ * This code is borrowed from https://www.figma.com/plugin-docs/working-with-images/
+ */
+export const transformWebpToPNG = async (bytes: Uint8Array) => {
```
Comment (samijaber) [inline plugin/functions/encode-images.ts:5]: This is the actual fix @steve8708 😄

File: plugin/ui.tsx [hunk 2]
```diff
@@ -275,6 +282,7 @@ class App extends SafeComponent {
   editorScriptAdded = false;
   dataToPost: any;
 
+  // TODO: THIS IS UNUSED
   async getImageUrl(
```
Comment (samijaber) [inline plugin/ui.tsx:286]: Curious what this function was intended for: it calls `/upload` to upload images, but is never actually called

File: lib/html-to-figma/helpers/image.ts [hunk 3]
```diff
@@ -1,14 +1,58 @@
...
+      type: "IMAGE",
+      // TODO: object fit, position
+      scaleMode: computedStyle.objectFit === "contain" ? "FIT" : "FILL",
+      imageHash: null,
+    };
+
+    if (el instanceof HTMLImageElement) {
+      // we use `currentSrc` instead of `src` as that will be the accurate value in dynamic contexts:
+      // when the img is a child of a picture element, or it has `sizes`/`srcSet` attributes, etc.
+      const url = el.currentSrc;
```
Comment (samijaber) [inline lib/html-to-figma/helpers/image.ts:34]: This is the other main improvement.

Comment (steve8708) [reply lib/html-to-figma/helpers/image.ts:34]: great find

File: plugin/functions/encode-images.ts (hunk 1 above)
Comment (steve8708) [reply plugin/functions/encode-images.ts:5]: awesome

File: plugin/ui.tsx (hunk 2 above)
Comment (steve8708) [reply plugin/ui.tsx:286]: ah - previous experiment that instead moved elsewhere. can be removed if we like

**Pull request merged**